### PR TITLE
Add a note about Python GIL

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -549,6 +550,22 @@ properly set the `--shm-size` flag depending on the size of your inputs and
 outputs. The default value for docker run command is `64MB` which is very
 small.
 
+## Python Global Interpreter Lock (GIL)
```
Comment (tanmayv25) [inline README.md:553]: "Multiple Model Instance Support"?
Then we can talk about how GIL prevents us from attaining concurrency in single python process.

File: README.md [hunk 2]
```diff
@@ -549,6 +550,22 @@ properly set the `--shm-size` flag depending on the size of your inputs and
...
+
+Python interpreter uses a global lock known as
+[GIL](https://docs.python.org/3/c-api/init.html#thread-state-and-the-global-interpreter-lock).
+Because of GIL, it is not possible have multiple threads running in the same
+Python interpreter simultaneously as each thread requires to acquire the GIL
+when accessing Python objects which will serialize all the operations. In order
+to work around this issue, Python backend spawns a complete separate process for each
```
Comment (CoderHam) [inline README.md:560]: Remove complete. Python backend spawns a separate process

File: README.md [hunk 3]
```diff
@@ -549,6 +550,22 @@ properly set the `--shm-size` flag depending on the size of your inputs and
...
+[model instance](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#multiple-model-instances).
+This is in contrast with how other Triton backends such as
+[ONNXRuntime](https://github.com/triton-inference-server/onnxruntime_backend),
+[Tensorflow](https://github.com/triton-inference-server/tensorflow_backend), and
```
Comment (CoderHam) [inline README.md:564]: TensorFlow

File: README.md [hunk 4]
```diff
@@ -549,6 +550,22 @@ properly set the `--shm-size` flag depending on the size of your inputs and
...
+[PyTorch](https://github.com/triton-inference-server/pytorch_backend) handle
+multiple instances. Increasing the instance count for these backends will create
+an additional thread instead of spawning a separate process.
```
Comment (CoderHam) [inline README.md:567]: create additional threads instead of spawning separate processes.

**Review by CoderHam — CHANGES_REQUESTED** [review]

**Review by CoderHam — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**